ddtrace/internal/ci_visibility/telemetry/events.py

```diff
@@ -0,0 +1,105 @@
+from enum import Enum
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from ddtrace.internal.ci_visibility.telemetry.constants import CIVISIBILITY_TELEMETRY_NAMESPACE as _NAMESPACE
+from ddtrace.internal.ci_visibility.telemetry.constants import EVENT_TYPES
+from ddtrace.internal.ci_visibility.telemetry.constants import TEST_FRAMEWORKS
+from ddtrace.internal.ci_visibility.telemetry.utils import skip_if_agentless
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.telemetry import telemetry_writer
+
+
+log = get_logger(__name__)
+
+
+class EVENTS_TELEMETRY(str, Enum):
+    CREATED = "event_created"
+    FINISHED = "event_finished"
+    MANUAL_API_EVENT = "manual_api_event"
+    ENQUEUED_FOR_SERIALIZATION = "events_enqueued_for_serialization"
+
+
+@skip_if_agentless
+def _record_event(
+    event: EVENTS_TELEMETRY,
+    event_type: EVENT_TYPES,
+    test_framework: Optional[TEST_FRAMEWORKS],
+    has_codeowners: Optional[bool] = None,
+    unsupported_ci: Optional[bool] = None,
+    is_benchmark: Optional[bool] = None,
+):
+    if has_codeowners is not None and event_type != EVENT_TYPES.SESSION:
+        log.debug("has_codeowners tag can only be set for sessions, but event type is %s", event_type)
+    if unsupported_ci is not None and event_type != EVENT_TYPES.SESSION:
+        log.debug("unsupported_ci tag can only be set for sessions, but event type is %s", event_type)
+    if is_benchmark is not None and event_type != EVENT_TYPES.TEST:
+        log.debug("is_benchmark tag can only be set for tests, but event type is %s", event_type)
+
+    _tags: List[Tuple[str, str]] = [("event_type", event_type)]
+    if test_framework and test_framework != TEST_FRAMEWORKS.MANUAL:
+        _tags.append(("test_framework", test_framework))
+    if event_type == EVENT_TYPES.SESSION:
+        if has_codeowners is not None:
```

Comment:
technically this could be `has_codeowners=False` (based on typing).

Should you do `has_codeowners: bool = False` instead? not sure why `Optional[bool]` would be needed here